tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+
+    // Every package has a name. Some packages have a homepage.
+    Console.info("Package:", data.name);
+    if (data.homepage) {
+      Console.info("Homepage:", Console.url(data.homepage));
+    }
+    // Git is per-version, so we will print the latest one, if one exists.
+    if (defaultVersion && defaultVersion.git) {
+      Console.info("Git:", Console.url(defaultVersion.git));
+    }
+    // Local packages might not have any maintainers.
+    if (! _.isEmpty(data.maintainers)) {
+      Console.info("Maintainers:", data.maintainers.join(", "));
+    }
+    Console.info();
+
+    // If we don't have a long-form description, we will use the summary. For a
+    // local package, we might not have a summary, in which case we should be
+    // careful not to print extra lines.
+    var printDescription = defaultVersion &&
+      (defaultVersion.description || defaultVersion.summary);
+    if (printDescription) {
+      Console.info(printDescription );
+      Console.info();
+    }
+
+    var totalShownVersions = 0;
+    // If we have any versions to show, print them out now.
+    if (data.versions && ! _.isEmpty(data.versions)) {
+      var versionsHeader =
+            self.showHiddenVersions ? "Versions:" : "Recent versions:";
+      Console.info(versionsHeader);
+      var rows = [];
+      _.each(data.versions, function (v) {
+        totalShownVersions++;
+
+        // For a local package, we don't have a published date, and we don't
+        // need to show if it has already been downloaded (it is local, we don't
+        // need to download it). Instead of showing both of these values, let's
+        // show the directory.
+        if (v.local) {
+          rows.push([v.version, v.directory]);
+          return;
+        }
+
+        // Convert the date into a display-friendly format, or print nothing for
+        // a local package.
+        var publishDate = utils.longformDate(v.publishedOn);
+
+        // If there is a status that we would like to report for this package,
+        // figure it out now.
+        if (v.installed) {
+          var paddedDate = padLongformDate(publishDate);
+          rows.push([v.version, paddedDate + "  " + "installed"]);
+        } else {
+          rows.push([v.version, publishDate]);
+        }
+      });
+      // The only time that we are going to go over a reasonable character limit
+      // is with a directory for the local package. We would much rather display
+      // the full directory than trail it off.
+      Console.printTwoColumns(rows, { indent: 2, ignoreWidth: true });
+    }
+
+    // If we have not shown all the available versions, let the user know.
+    if (data.totalVersions > totalShownVersions) {
+      var versionsPluralizer =
+            (data.totalVersions > 1) ?
+            "all " + data.totalVersions + " versions" :
+            "the hidden version";
+      Console.info(
+        "Some versions of", self.name, "have been hidden.",
```

Comment:
I think that last suggestion sounds reasonable.  I dunno, my reaction to the current message really is "WHYYYY?????" :)